client/src/lib/firebase/authfunctions.ts

```diff
@@ -0,0 +1,42 @@
+import { app } from "../../firebase";
+import {
+  getAuth,
+  GoogleAuthProvider,
+  OAuthProvider,
+  signInWithPopup,
+  signInAnonymously,
+  signInWithEmailAndPassword,
+} from "firebase/auth";
+
+// Google login/signup
+export async function loginWithGoogle() {
+  const auth = getAuth(app);
+  const provider = new GoogleAuthProvider();
+  signInWithPopup(auth, provider).then((result) => {
+    // This gives you a Google Access Token. You can use it to access the Google API.
+    const credential = GoogleAuthProvider.credentialFromResult(result);
+    const token = credential!.accessToken;
+    // The signed-in user info.
+    const user = result.user;
+  });
```

Comment:
it looks like we are not using any of these values so we probobly don't need the then block